setup.py

```diff
@@ -169,29 +221,42 @@
         " specify it using the environment variable FMIL_HOME."
     )
 
-if 0 != sys.argv[1].find("clean"): #Dont check if we are cleaning!
-
-    #Check to see if FMILIB_SHARED exists and if so copy it, otherwise raise exception
-    if sys.platform.startswith("win"):
-        dirs_to_search = libdirs + bindirs
-        while dirs_to_search:
-            path_to_dir = os.path.abspath(dirs_to_search.pop())
-            for file_name in os.listdir(path_to_dir):
-                full_path = os.path.join(path_to_dir, file_name)
-                if "fmilib_shared" in file_name and not file_name.endswith(".a"):
-                    fmilib_shared = shutil.copy2(full_path, os.path.join(".", "src", "pyfmi"))
-                    dirs_to_search = None
-                    break
-        if not fmilib_shared:
-            raise Exception(
-                f"Could not find shared library 'fmilib_shared' at either location:" + \
-                    f"\n\t{', '.join(dirs_to_search)}")
-
-        if copy_gcc_lib:
-            path_gcc_lib = ctypes.util.find_library("libgcc_s_dw2-1.dll")
-            if path_gcc_lib is not None:
-                shutil.copy2(path_gcc_lib,os.path.join(".","src","pyfmi"))
-                gcc_lib = os.path.join(".","src","pyfmi","libgcc_s_dw2-1.dll")
+def find_dynamic_fmil_library():
+    """ Find the dynamic library of FMIL. """
+    dirs_to_search = libdirs + bindirs
+    for path_to_dir in dirs_to_search:
+        path_to_dir = os.path.abspath(path_to_dir)
+
+        if not os.path.exists(path_to_dir):
+            continue
+
+        for file_name in os.listdir(path_to_dir):
+            full_path = os.path.join(path_to_dir, file_name)
+            if fmil_name in file_name and not file_name.endswith(".a"):
+                return full_path
+
+    raise Exception(
+        f"Could not find shared library '{fmil_name}' at either locations:" + \
+            f"\n\t{', '.join(dirs_to_search)}")
+
+if 0 != sys.argv[1].find("clean"): # Dont check if we are cleaning!
+
+    use_dynamic_fmil_library = fmil_name.endswith("shared") # this should be improved in a future release
```

Comment:
```suggestion
    use_dynamic_fmil_library = fmil_name.endswith("shared") # TODO: this should be improved in a future release
```
Without a TODO or similar, the chances of finding this are slim.

Reply:
Agree, fixed